### PR TITLE
[Handwired] Support for Adafruit lemon keypad

### PR DESCRIPTION
**Pull request by dari-studios**
## Description

QMK + Via support for a handwired Lemon-themed keypad, designed by Adafruit. Modified for a pro micro.
https://learn.adafruit.com/qtpy-lemon-mechanical-keypad-macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
This contains changes from other PRs, which need to be reverted.

File: keyboards/handwired/lemonpad/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    	[_EMPTY7] = LAYOUT(
+        		KC_TRNS,    KC_TRNS,    KC_TRNS,
+        		KC_TRNS,    KC_TRNS,    KC_TRNS
+    	),
+
+    	[_EMPTY8] = LAYOUT(
+        		KC_TRNS,    KC_TRNS,    KC_TRNS,
+        		KC_TRNS,    KC_TRNS,    KC_TRNS
+    	)
```
Comment (drashna) [inline keyboards/handwired/lemonpad/keymaps/default/keymap.c:77]:
```suggestion
```

File: keyboards/handwired/lemonpad/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+enum layer_names {
+    	_BASE,
+    	_EMPTY1,
+    	_EMPTY2,
+    	_EMPTY3,
+    	_EMPTY4,
+    	_EMPTY5,
+    	_EMPTY6,
+    	_EMPTY7,
+    	_EMPTY8
```
Comment (drashna) [inline keyboards/handwired/lemonpad/keymaps/default/keymap.c:30]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/lemonpad/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# lemonpad
+
+![lemonpad](https://i.imgur.com/hAp2Nic.jpeg)
```
Comment (fauxpark) [inline keyboards/handwired/lemonpad/readme.md:3]:
```suggestion
![lemonpad](https://i.imgur.com/hAp2Nich.jpeg)
```

File: keyboards/handwired/lemonpad/readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+This is a QMK and VIA implementation for an atmega32u4 based adaptation of the Adafruit Lemon Keypad. By default it uses a Pro Micro, if you would like to use a different microcontroller please make sure you have the correct bootloader set. 
+
+* Keyboard Maintainer: [dari-studios](https://github.com/dari-studios)
+* Hardware Supported: atmega32u4 based boards with the Caterina bootloader
+* Hardware Availability: https://learn.adafruit.com/qtpy-lemon-mechanical-keypad-macropad
+* VIA JSON: https://pastebin.com/WXR3jy52
```
Comment (fauxpark) [inline keyboards/handwired/lemonpad/readme.md:10]:
```suggestion
```

File: keyboards/handwired/lemonpad/config.h [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x6473
+#define PRODUCT_ID   0x4C50
+#define DEVICE_VER   0x0001
+#define MANUFACTURER dari-studios
+#define PRODUCT      dari-studios lemonpad
```
Comment (fauxpark) [inline keyboards/handwired/lemonpad/config.h:26]:
```suggestion
#define PRODUCT      lemonpad
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by dari-studios** [comment]
Reverted changes from other PRs and added suggested changes. Thank you all for your time.

**Comment by drashna** [comment]
Other changes are still there, it looks like.  Rebasing may work better to fix the issues.

File: keyboards/handwired/lemonpad/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+VIA_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/handwired/lemonpad/rules.mk:10]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by zvecr — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**